contracts/ForkableRealityETH_ERC20.sol

```diff
@@ -2,977 +2,173 @@
-        questions[question_id].opening_ts = opening_ts;
-        questions[question_id].timeout = timeout;
     }
 
-    /// @notice Submit an answer for a question.
-    /// @dev Adds the answer to the history and updates the current "best" answer.
-    /// May be subject to front-running attacks; Substitute submitAnswerCommitment()->submitAnswerReveal() to prevent them.
-    /// @param question_id The ID of the question
-    /// @param answer The answer, encoded into bytes32
-    /// @param max_previous If specified, reverts if a bond higher than this was submitted after you sent your transaction.
-    /// @param tokens The amount of tokens to submit
-    function submitAnswerERC20(
-        bytes32 question_id,
-        bytes32 answer,
-        uint256 max_previous,
-        uint256 tokens
-    )
-        external
-        stateOpen(question_id)
-        bondMustDouble(question_id, tokens)
-        previousBondMustNotBeatMaxPrevious(question_id, max_previous)
-    {
-        _deductTokensOrRevert(tokens);
-        _addAnswerToHistory(question_id, answer, msg.sender, tokens);
-        _updateCurrentAnswer(question_id, answer);
-    }
+    function _createInitialTemplates() internal override {
+        // TODO: Decide if we want to include the original templates for consistency/flexibility
+        // ...even though they won't be used unless we upgrade the forkmanager or whatever governs this.
 
-    /// @notice Submit an answer for a question, crediting it to the specified account.
-    /// @dev Adds the answer to the history and updates the current "best" answer.
-    /// May be subject to front-running attacks; Substitute submitAnswerCommitment()->submitAnswerReveal() to prevent them.
-    /// @param question_id The ID of the question
-    /// @param answer The answer, encoded into bytes32
-    /// @param max_previous If specified, reverts if a bond higher than this was submitted after you sent your transaction.
-    /// @param answerer The account to which the answer should be credited
-    /// @param tokens The amount of tokens to submit
-    function submitAnswerForERC20(
-        bytes32 question_id,
-        bytes32 answer,
-        uint256 max_previous,
-        address answerer,
-        uint256 tokens
-    )
-        external
-        payable
-        stateOpen(question_id)
-        bondMustDouble(question_id, tokens)
-        previousBondMustNotBeatMaxPrevious(question_id, max_previous)
-    {
-        _deductTokensOrRevert(tokens);
-        require(answerer != NULL_ADDRESS, "answerer must be non-zero");
-        _addAnswerToHistory(question_id, answer, answerer, tokens);
-        _updateCurrentAnswer(question_id, answer);
+        // Bump the next template ID to stay clear of the range of the normal stock templates, as this might confuse the UI
+        nextTemplateID = UPGRADE_TEMPLATE_ID;
+        createTemplate(
+            '{"title": "Should we execute the upgrade contract %s?", "type": "bool"}'
```

Comment:
wouldn't we say how its being upgraded? E.g. to which git commit ?

Reply:
I think it would either be "execute the code with this hash" or "call this contract". A git commit won't work because the contract code can't read git. Will take a look at how safe/snapshot does it.